Foundation/String/UTF8.hs

```diff
@@ -820,6 +822,76 @@ reverse s@(String ba) = runST $ do
                 _  -> return () -- impossible
             loop ms (si `offsetPlusE` nb) d
 
+-- Finds how many times we need to perform the replacement, and where are
+-- the insertion points.
+indices :: String -> String -> (Int, [Offset8])
+indices needle@(String ned) haystack@(String hy) =
+  case haystackLen < needleLen of
+    True  -> (0, [])
```

Comment:
needleLen of 0 will result in an infinite loop - it should raise an error saying 0 length is disallowed.